### ISSUE TITLE
Add feature that custom `created_at` and `updated_at` column name in database

### ISSUE THREAD

**Issue by JunQiGenYuan** [strapi/strapi#998]
https://github.com/strapi/strapi/blob/6b8f430872b248dcc7510c9a540fbd55c50116f3/packages/strapi-bookshelf/lib/index.js#L90

If change line above to `hasTimestamps: _.get(definition, 'options.timestamps'),`, remove logic equal to `true`.
Then can set the [model].settings.json `options.timestamps` as an array like this `["createTime", "updateTime"]` to add a feature that custom `created_at` and `updated_at`.

Check the bookshelf.js github:
https://github.com/bookshelf/bookshelf/blob/880f3bf17f3557fe2c491866aaa2c5694eb65577/src/base/model.js#L615

**Comment by lauriejim** [comment]
Hello @JunQiGenYuan this feature is not a priority for us, but fill free to submit a PR !

**Comment by derrickmehaffy** [comment]
@JunQiGenYuan any update on the PR? Just checking to see if we can close out old issues.

**Comment by lauriejim** [comment]
Hello @JunQiGenYuan we’ll appreciate your contribution on this issue! Did you have time to work on this  and submit a PR to share it with the community ?

Check out the contributing guide to get started: https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md

**Comment by chaitanyya** [comment]
@lauriejim This is something we'd like to have. I'd really like to pick this up.

**Comment by lauriejim** [comment]
Reminder for this feature: have to work for bookshelf and mongoose.

**Comment by derrickmehaffy** [comment]
I mentioned to @chaitanyya on slack for manual testing to be done on both hooks @lauriejim

**Comment by chaitanyya** [comment]
Things work as expected when I use the correct timestamp attributes in this file instead of the hard coded values.
https://github.com/strapi/strapi/blob/e4dc27be54f36a956a06d35cab5f1ae11f2b9836/packages/strapi-plugin-content-manager/admin/src/containers/EditPage/saga.js#L79-L89

I'll need access to the model fields. I see the object that's stored as `db_model_<modelName>` in database have field type values. Maybe using that to determine the correct timestamp attributes would be nice.

**Comment by lauriejim** [comment]
Hello @chaitanyya thank you for this feedback.
Actually a large part of this topic is hard coded you are right!

I think we have to expose the same API if you are using Mongoose or Bookshelf.
That impact 2 thing:
- `_id` key for Mongoose (we have to remove it and have only `id`)
- The `created` and `updated` at key format.

I would like @Aurelsicoko feedback about this topic (he will come back 02/01/2019)

**Comment by chaitanyya** [comment]
Yes, I checked the fields object in schema it doesn't have primaryKey and Timestamps.

https://github.com/strapi/strapi/blob/e4dc27be54f36a956a06d35cab5f1ae11f2b9836/packages/strapi-plugin-content-manager/admin/src/containers/EditPage/saga.js#L95

Maybe having them available in the fields, we can then add a check on the type of the field and delete the `created` and `updated` values instead of hardcoding them. Correct me if I'm going in the wrong direction.

**Comment by chaitanyya** [comment]
Hey guys -

I was able to expose the timestamps and use the user defined values to remove correct attributes from the `record` object. 

**There's still some work that needs to be done, here's a dough list of what I believe needs to be taken care of:** 

- Default values if no attributes are provided.
- Update documentation to explain how it can be done.
- Ability to define them in the admin UI (Maybe a separate feature)
- Schema cleanup (Maybe changing the attribute name for it)

I'd love some feedback on the method that's currently used in making this feature available.

**Comment by lauriejim** [comment]
Hello @chaitanyya I will try to check your stuff this week (I hope to have time to do it)

**Comment by chaitanyya** [comment]
*Update:*

- Uses default if nothing is provided.
- Works with existing model settings configuration (Will do some more tests later this week)

**Comment by chaitanyya** [comment]
The PR is merged and it should be usable in the coming release. Can we please close this issue now?
